src/log_batch.rs

```diff
@@ -728,7 +729,7 @@ impl LogBatch {
 
     /// Adds a protobuf key value pair into the log batch.
     pub fn put_message<S: Message>(&mut self, region_id: u64, key: Vec<u8>, s: &S) -> Result<()> {
-        if crate::is_internal_key(&key, None) {
+        if crate::is_internal_key(&key, Some(ATOMIC_GROUP_KEY)) {
```

Comment:
Don't change the callsite. Implement two versions of `is_internal_key` instead:

For `cfg(test)`, keep the behavior the same as before.

For `not(cfg(test))`, we only check `ATOMIC_GROUP_KEY` even for `is_internal_key(key, None)`. And we assert all key prefix passed to `make_internal_key` is `ATOMIC_GROUP_KEY`.